modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -93,37 +112,98 @@
+        return Sets.newHashSet(getMappedPort(REST));
     }
 
     @Override
-    public Set<Integer> getLivenessCheckPortNumbers() {
-        return Sets.newHashSet(getLivenessCheckPort());
+    @SneakyThrows
+    protected void doStart() {
+        String networkAlias = getNetworkAliases().get(0);
+        startProxy(networkAlias);
+
+        for (CouchbasePort port : CouchbasePort.values()) {
+            exposePortThroughProxy(networkAlias, port.getOriginalPort(), getMappedPort(port));
+
+            // CAPI port has a special configuration file which is set via env
+            if (port == CAPI) {
+                withEnv(port.getName(), getMappedPort(port) + "");
+            }
+        }
+        super.doStart();
+    }
+
+    private void startProxy(String networkAlias) {
+        proxy = new SocatContainer().withNetwork(getNetwork());
+
+        for (CouchbasePort port : CouchbasePort.values()) {
+            if (port.isBootstrap()) {
+                proxy.withTarget(port.getOriginalPort(), networkAlias);
+            } else {
+                proxy.addExposedPort(port.getOriginalPort());
+            }
+        }
+        proxy.setWaitStrategy(null);
+        proxy.start();
+    }
+
+    private void exposePortThroughProxy(String networkAlias, int originalPort, int mappedPort) {
+        ExecCreateCmdResponse createCmdResponse = dockerClient
+            .execCreateCmd(proxy.getContainerId())
+            .withCmd("/usr/bin/socat", "TCP-LISTEN:" + originalPort + ",fork,reuseaddr", "TCP:" + networkAlias + ":" + mappedPort)
+            .exec();
+
+        dockerClient.execStartCmd(createCmdResponse.getId())
+            .exec(new ExecStartResultCallback());
+    }
+
+    @Override
+    public List<Integer> getExposedPorts() {
+        return proxy.getExposedPorts();
+    }
+
+    @Override
+    public String getContainerIpAddress() {
+        return proxy.getContainerIpAddress();
     }
 
     @Override
-    protected void configure() {
-        // Configurable ports
-        addExposedPorts(11210, 11207, 8091, 18091);
+    public Integer getMappedPort(int originalPort) {
+        return proxy.getMappedPort(originalPort);
+    }
 
-        // Non configurable ports
-        addFixedExposedPort(8092, 8092);
-        addFixedExposedPort(8093, 8093);
-        addFixedExposedPort(8094, 8094);
-        addFixedExposedPort(8095, 8095);
-        addFixedExposedPort(18092, 18092);
-        addFixedExposedPort(18093, 18093);
-        setWaitStrategy(new HttpWaitStrategy().forPath("/ui/index.html#/"));
+    protected Integer getMappedPort(CouchbasePort port) {
+        return getMappedPort(port.getOriginalPort());
+    }
+
+    @Override
+    public List<Integer> getBoundPortNumbers() {
+        return proxy.getBoundPortNumbers();
+    }
+
+    @Override
+    public void stop() {
+        Stream.<Runnable>of(super::stop, proxy::stop, this::stopCluster).parallel().forEach(Runnable::run);
+    }
+
+    private void stopCluster() {
```

Comment:
what's the point of it? We're forcibly shutting down the container, isn't it enough?